File: android/src/main/java/com/socialauth/SocialAuthResponse.kt
package com.socialauth

import android.util.Log
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

data class SocialAuthResponse(
  val token: String,
  val firstName: String,
  val lastName: String,
  val email: String,
  val phone: String,
  val imageUrl: String
) {

  fun toMap(): WritableMap {
    val map = Arguments.createMap()

    map.putString("token", this.token)
    map.putString("firstName", this.firstName)
    map.putString("lastName", this.lastName)
    map.putString("email", this.email)
    map.putString("phone", this.phone)
    map.putString("imageUrl", this.imageUrl)

    if (BuildConfig.DEBUG) {
      Log.w("REVault", map.toString())
    }

    return map
  }
}

File: ios/SocialAuthResponse.swift
//
//  SocialAuthResponse.swift
//  social-auth
//
//  Created by William Connelly on 30/3/21.
//

import Foundation

struct SocialAuthResponse {
    
    let token: String
    let firstName: String
    let lastName: String
    let email: String
    let phone: String
    let imageUrl: String
    
    var toDictionary: [String: Any] {
        return [
            "token": self.token,
            "firstName": self.firstName,
            "lastName": self.lastName,
            "email": self.email,
            "phone": self.phone,
            "imageUrl": self.imageUrl
        ]
    }
}
